Refactored getMetadata method of Asset to include unused parameters (#12800)
pimcore/pimcore: doc/Development_Documentation/04_Assets/01_Working_with_PHP_API.md

### Examples
##### Getting Data

> Metadata for an asset can be fetched using `getMetadata` method of Asset class. It takes 4 parameters: 

Reviewer: ```suggestion
Metadata for an asset can be fetched using `getMetadata` method of Asset class. It can take the following optional parameters: 
```
saying `optional` should be sufficent to hint that all params can be `nullable`

> 1. `$name` can be a nullable string which corresponds to the name of metadata. If null is passed, it tries to get all the metadata. The actual contents may be manipulated by other parameters.

Reviewer: ```suggestion
1. `$name` (string) when passed it returns the metadata with the given name, otherwise it lists all.
```

this should implicitly suggest that it can either return a single value or a set of values (array)

> 2. `$language` is a nullable string which corresponds to language of the metadata. If language is passed, it tries to fetch metadata in a specific language.

Reviewer: ```suggestion
2. `$language` (string) with this parameter you can filter out the metadata of a specific language.
```

> 3. `$strictMatch` is a boolean value which, if true tries to get only the metadata which exactly matches the requested language.

Reviewer: ```suggestion
3. `$strictMatch` (boolean) if true tries to get only the metadata which exactly matches the requested language without considering the fallback.
```

> 3. `$strictMatch` is a boolean value which, if true tries to get only the metadata which exactly matches the requested language.

Reviewer 2: Maybe this parameter can be renamed to better reflect what is does, like `$strictLanguageMatch`.

Reviewer: I agree, without specifying language, it could be "confused" for some strict matching metadata name feature, like if it would be possible to fuzzy search eg. `copy*`.

Author: Renamed the flag to `$strictMatchLanguage`. Thanks @jdreesen for the feedback!

> 4. `$raw` is a boolean value which, if true gets extra information like name, data, language and input type of the metadata.

Reviewer: ```suggestion
4. `$raw`  (boolean) if true, it will also return extra information like name, data, language and input type of the metadata.
```

// get the English title
$asset->getMetadata("title", "en");
// if there's no title for "en" but one without a language this will be returned (fallback mechanism).

// get all available metadata
$asset->getMetadata();

// get all available metadata for a specific language along with metadata which have no language assigned
$asset->getMetadata(null, 'en');

> // get all available metadata for a specific language discarding any that does not have language set

Reviewer: ```suggestion
// get exclusively all the metadata that have a specific language
```
